carbon-design-system/carbon
# fix(contextual-layout-tokens): support individual component style imports

Ref #13968 

Closes #13986

#### Changelog

**Changed**

- Imported layout module via `@use` from `/styles/utilities/layout.scss` to ensure tokens are always present when needed
- Added ContainedList styles to build files
- Added fallbacks for type properties used in `contained-list.scss` and `button/mixin.scss`
- Improve isolated rendering of ContainedList (when only `contained-list.scss` is imported by the user)

#### Testing / Reviewing

Replace `packages/react/.storybook/styles.scss` with the following:
```scss
@use '../scss/components/accordion';
@use '../scss/components/button';
@use '../scss/components/contained-list';
@use '../scss/components/content-switcher';
@use '../scss/components/copy-button';
@use '../scss/components/progress-bar';
@use '../scss/components/tag';
@use '../scss/components/text-input';
@use '../scss/components/tile';
```

- Verify the components supporting contextual layout tokens don't have any layout token-related regressions when imported without global styles:
  - Accordion
  - Button
  - ContainedList
  - ContentSwitcher
  - CopyButton
  - ProgressBar
  - Tag
  - TextInput
  - Tile
- Verify the layout tokens are only emitted once in the generated `packages/styles/css/styles.css`

Note: With the isolated styles there are regressions in color caused by missing imports of the `layer` module. To contain the size of PRs this is not covered in this PR but should be handled in a separate one.
